src/main/scala/rifs/business/controllers/ApplicationController.scala

```diff
@@ -46,8 +47,77 @@ class ApplicationController @Inject()(val cached: Cached, applications: Applicat
+  def deleteSectionItem(id: ApplicationId, sectionNumber: Int, itemNumber: Int) = Action.async { implicit request =>
+    def hasItemNumber(o: JsObject, num: Int) = o \ "itemNumber" match {
+      case JsDefined(JsNumber(n)) if n == num => true
+      case _ => false
+    }
+
+    applications.fetchAppWithSection(id, sectionNumber).flatMap {
+      case Some((app, os)) =>
+        val doc = os.map(_.answers).getOrElse(JsObject(Seq()))
+        val items = doc \ "items" match {
+          case JsDefined(JsArray(is)) => is.collect { case o: JsObject => o }
+          case _ => Seq()
+        }
+        val remainingItems = items.filterNot(o => hasItemNumber(o, itemNumber))
+        val updated = doc + ("items" -> JsArray(remainingItems))
+        applications.saveSection(id, sectionNumber, updated).map(_ => NoContent)
+
+      case None => Future.successful(NotFound)
+    }
+  }
+
+  def putSectionItem(id: ApplicationId, sectionNumber: Int, itemNumber: Int) = Action.async(parse.json[JsObject]) { implicit request =>
+    def hasItemNumber(o: JsObject, num: Int) = o \ "itemNumber" match {
+      case JsDefined(JsNumber(n)) if n == num => true
+      case _ => false
+    }
+
+    applications.fetchAppWithSection(id, sectionNumber).flatMap {
+      case Some((app, os)) =>
+        val doc = os.map(_.answers).getOrElse(JsObject(Seq()))
+        val items = doc \ "items" match {
+          case JsDefined(JsArray(is)) => is.collect { case o: JsObject => o }
+          case _ => Seq()
+        }
+        val remainingItems = items.filterNot(o => hasItemNumber(o, itemNumber))
+        val newItem = request.body + ("itemNumber" -> JsNumber(itemNumber))
+        val updated = doc + ("items" -> JsArray(newItem +: remainingItems))
+        applications.saveSection(id, sectionNumber, updated).map(_ => NoContent)
+
+      case None => Future.successful(NotFound)
+    }
+  }
+
+  def postSectionItem(id: ApplicationId, sectionNumber: Int) = Action.async(parse.json[JsObject]) { implicit request =>
+    applications.fetchAppWithSection(id, sectionNumber).flatMap {
+      case Some((app, os)) =>
+        val doc = os.map(_.answers).getOrElse(JsObject(Seq()))
+
+        val items: Seq[JsValue] = doc \ "items" match {
+          case JsDefined(JsArray(is)) => is
+          case _ => Seq()
+        }
+
+        allocateItemNumber().flatMap { n =>
+          val newItem = request.body + ("itemNumber" -> JsNumber(n))
+          val updated = doc + ("items" -> JsArray(newItem +: items))
+
+          applications.saveSection(id, sectionNumber, updated).map(_ => NoContent)
+        }
+
+      case None => Future.successful(NotFound)
+    }
+  }
+
+  def allocateItemNumber(): Future[Int] = Future {
+    Random.nextInt().abs
+  }
+
+  def completeSection(id: ApplicationId, sectionNumber: Int) = Action.async(parse.json[JsObject]) {
+    implicit request =>
+      applications.saveSection(id, sectionNumber, request.body, Some(LocalDateTime.now())).map(_ => NoContent)
```

Comment:
Should we always use UTC instead? (We're unlikely to run this code outside the UK, but still)